core/etcd/config/templates/stack-template.json

```diff
@@ -496,7 +496,12 @@
                   "'\n",
                   "ETCD_VERSION='",
                     "{{$.Etcd.Version}}",
-                  "'\n"
+                  "'\n",
+                  {{if $.Etcd.FormatOpts}}
+                  "ETCD_OPTS='",
+                    "{{$.Etcd.FormatOpts}}",
+                  "'\n",
+                  {{end}}
```

Comment:
Perhaps you have an extra `,` after the last element of the format opts, that is causing this error?

```
    	maincluster_test.go:3533: failed to create cluster driver : failed to initialize etcd stack: Error while rendering template: invalid character ']' looking for beginning of value:
    		json syntax error (offset=10035), in this region:
    		-------
    		,
    		                    "--quota-backend-bytes=2147483648",
    		                  "'\n",
    		                  
    		                ]]}
    		              }
    		            }
    		          }
    		
    		-------
```